packages/chat-component-bindings/src/hooks/usePropsFor.ts

```diff
@@ -78,12 +80,27 @@ export const getSelector = <Component extends (props: any) => JSX.Element | unde
   return findSelector(component);
 };
 
+const messageThreadSelectorsByThread = {};
+
 const findSelector = (component: (props: any) => JSX.Element | undefined): any => {
+  // For the message thread selector we need to create a new one for each thread
+  // If we have just one for the entire app, then we will have updates when not expecting due to
+  // the arguments changing
+  const getMessageThreadSelector: () => MessageThreadSelector = () => {
+    const threadId = useContext(ChatThreadClientContext)?.threadId ?? 'single-chat-thread-default-id';
```

Comment:
The default one will only kick in when there is no thread, so let's call it default-id-for-no-thread or something? Cause even single thread it will follow the thread id you have